### ISSUE TITLE
`src\EntryPoint\Exceptions\UserFacingExceptionDefaults.cs` breaks unit tests

### ISSUE THREAD

**Issue by NickStrupat** [Nick-Lucas/EntryPoint#49]
Hi, thank you for your work on this library. It has helped us build a clean command line program.

Waiting on the `Console.ReadLine()` is breaking my unit tests. They just hang there. Since no output is captured by xUnit, it was laborious to find the problem.

https://github.com/Nick-Lucas/EntryPoint/blob/6f59c1769cb80c897757b8a3b7b3b76ee0514f44/src/EntryPoint/Exceptions/UserFacingExceptionDefaults.cs#L17

Would you consider removing the `ReadLine` call?

**Comment by Nick-Lucas** [comment]
Hi Nick,

The idea with this method is that it's a sensible default for most applications; you can override it on your own implementation with an implementation more appropriate for you.

Please see https://github.com/Nick-Lucas/EntryPoint/blob/6f59c1769cb80c897757b8a3b7b3b76ee0514f44/src/EntryPoint/BaseCliArguments.cs#L55

**Comment by Nick-Lucas** [comment]
I'm open to any suggestions of course, and I've seen these invisible test failures/hangs while testing EntryPoint itself. For now I'm marking `won't fix` but I'll have a think on it.

Something I have been considering is stripping back EntryPoint for a v2, to remove all framework like behaviours where it tries to take control of your application. That would mean removing the virtual methods and letting the developer manually check for flags like "HelpInvoked" and deal with them. It would fix a lot of 'default behaviour' case, but I'm in two minds on whether it's the right direction to go.

**Comment by NickStrupat** [comment]
Hi Nick,

That makes a ton of sense. Thanks for getting back to me so quickly. I can see how that's a tough decision. Sensible defaults for most uses are good, and good testability by default is also important in reducing dev time.
